### PR TITLE
Improve generation of test snippets

### PR DESCRIPTION
**Pull request by dataclouder**
When we run the test suite, we generate Terraform scripts that
can later be used for manual tests. The changes in this commit
add elements to the provider portion of the script that make the
manual test setup faster.

To use the improvement, run the tests with the environment variable `VCD_ADD_PROVIDER=1`, and then look into `./vcd/test-artifacts`

Signed-off-by: Giuseppe Maxia <gmaxia@vmware.com>

### CODE REVIEW

File: vcd/config_test.go [hunk 1]
```diff
@@ -74,19 +74,24 @@ type TestConfig struct {
 		UploadPieceSize int64  `json:"uploadPieceSize,omitempty"`
 		UploadProgress  bool   `json:"uploadProgress,omitempty"`
 	} `json:"media"`
+	EnvVariables map[string]string `json:"envVariables,omitempty"`
 }
 
 const (
+	// This library major version
+	currentProviderVersion = "2.0"
```
Comment (lvirbalas) [inline vcd/config_test.go]: We have a VERSION file - it should be reused here.

File: vcd/sample_vcd_test_config.json [hunk 2]
```diff
@@ -72,5 +72,12 @@
     "//": "Size in megabytes",
     "uploadPieceSize": 5,
     "uploadProgress": true
+  },
+  "envVariables" : {
+    "//" : "Environment variables that we want to set (or unset)",
```
Comment (lvirbalas) [inline vcd/sample_vcd_test_config.json]: It's fully clear, I suggest rephrasing to: "Environment variables that we want the test to set (or unset)"

File: vcd/sample_vcd_test_config.json [hunk 3]
```diff
@@ -72,5 +72,12 @@
...
+    "GOVCD_DEBUG": "",
+    "VCD_SKIP_TEMPLATE_WRITING": "",
+    "VCD_ADD_PROVIDER": ""
```
Comment (lvirbalas) [inline vcd/sample_vcd_test_config.json]: We are at a point where we need to document environment variables in the Terraform project. README.md needs to be updated or even better a separate TESTING.md created with all the nitty details of test setup.

File: vcd/config_test.go (hunk 1 above)
Comment (dataclouder) [reply vcd/config_test.go]: Done. Changed the code to get the version from there

File: vcd/sample_vcd_test_config.json (hunk 2 above)
Comment (dataclouder) [reply vcd/sample_vcd_test_config.json]: Done

File: vcd/sample_vcd_test_config.json (hunk 3 above)
Comment (dataclouder) [reply vcd/sample_vcd_test_config.json]: Removed testing instructions from `README.md` and added a `TESTING.md` file with explanation of the environment variables.

**Pull request closed without merging**